File: server/src/main/java/com/smartexpense/api/application/port/in/NotificationUseCase.java
package com.smartexpense.api.application.port.in;

import com.smartexpense.api.application.dto.response.NotificationResponseDto;
import java.util.List;
import java.util.UUID;

public interface NotificationUseCase {
    List<NotificationResponseDto> getNotifications(String email);
    void markAsRead(String email, UUID notificationId);
    void markAllAsRead(String email);
}

File: server/src/main/kotlin/com/smartexpense/api/application/port/in/NotificationUseCase.kt
package com.smartexpense.api.application.port.`in`

import com.smartexpense.api.application.dto.response.NotificationResponseDto
import java.util.UUID

interface NotificationUseCase {
    fun getNotifications(email: String): List<NotificationResponseDto>
    fun markAsRead(email: String, notificationId: UUID)
    fun markAllAsRead(email: String)
}
